Source/DafnyCore/Resolver/ModuleResolver.cs

```diff
@@ -792,9 +792,25 @@ public partial class ModuleResolver {
         new ActualBindings(f.Formals.ConvertAll(Expression.CreateIdentExpr)).ArgumentBindings,
         tok);
       var post = new AttributedExpression(new BinaryExpr(tok, BinaryExpr.Opcode.Eq, r, fn));
+      List<FrameExpression> reads;
+      if (Options.Get(CommonOptionBag.ReadsClausesOnMethods)) {
```

Comment:
That would make it hard/impossible to reject explicit reads clauses if the option is not turned on though, since you couldn't distinguish between code that explicitly includes `reads *`.

Reply:
Wouldn't you reject that code as part of parsing as well ?

Reply:
On revisiting this, yes, I think I could have assigned the default during parsing instead (with the same conditionals to avoiding doing that on lemmas etc.) I was wary of that at the time since I didn't see any other examples of adding default elements at parse time, and didn't want to risk more cases of errors that pointed to tokens that didn't exist in the original program, although in this particular case I don't think that's possible (since `reads *` can't cause errors AFAICT).